src/mds/FSMap.h

```diff
@@ -259,17 +259,21 @@ class FSMap {
       return;
     }
 
-    for (auto &f : filesystems) {
-      std::string_view fs_name = f.second->mds_map.get_fs_name();
+    for(auto it = filesystems.begin(); it != filesystems.end();) {
+      std::string_view fs_name = (*it).second->mds_map.get_fs_name();
       if (std::find(allowed.begin(), allowed.end(), fs_name) == allowed.end()) {
-	filesystems.erase(f.first);
+        it = filesystems.erase(it);
```

Comment:
The `filesystems` is a map, and there has no erase func will return an iterator, please see https://cplusplus.com/reference/map/map/erase/.

Reply:
Hi Xiubo @lxbsz ,

here we are iterating over map and an erasure of an element invalidates the iterator.
Therefore we need to have a valid iterator and we use erase.
The cppreference [https://en.cppreference.com/w/cpp/container/map/erase](https://en.cppreference.com/w/cpp/container/map/erase) explains:
_
Return value
1-2) Iterator following the last removed element. (since C++11)
_

Also you approved the fix last time (sorry, the PR was closed...):
[https://github.com/ceph/ceph/pull/45778](https://github.com/ceph/ceph/pull/45778)

Reply:
Hi Xiubo @lxbsz Thank you very much for the review!